source/path.js

```diff
@@ -2,7 +2,9 @@ import _curry2 from './internal/_curry2';
 import paths from './paths';
 
 /**
- * Retrieve the value at a given path.
+ * Retrieve the value at a given path. String `Idx` that could be parsed as
+ * non-negative integers has the same lookup behavior as their Integer
+ * counterparts. Non-negative integer `Idx` only work on an Array `obj`.
```

Comment:
While I agree that this needs some better explanation, I think this suggestion is too tied to your use-case, which I think is fairly unusual.  I'll try to think about it this weekend.

Reply:
How about something like this?

> Retrieves the value at a given path.  The nodes of the path can be arbitrary strings or non-negative integers.  For anything else, the value is unspecified.  Integer paths are meant to index arrays, strings are meant for objects.